Shared/Samples/Create KML multi-track/CreateKMLMultiTrackView.swift

```diff
@@ -0,0 +1,186 @@
+                                    state = newValue ? .recording : .navigating
+                                }
+                            )
+                            
+                            Spacer()
+                            
+                            Button("Save", systemImage: "square.and.arrow.down") {
+                                state = .viewingMultiTrack
+                            }
+                            .disabled(model.tracks.isEmpty)
+                        }
+                    }
+                }
+                .task(id: state) {
+                    // Runs the asynchronous action associated with the sample state.
+                    do {
+                        switch state {
+                        case .navigating:
+                            break
+                        case .recording:
+                            for await location in model.locationDisplay.$location where location != nil {
+                                model.addTrackElement(at: location!.position)
+                            }
+                            
+                            model.addTrack()
+                        case .viewingMultiTrack:
+                            await model.locationDisplay.dataSource.stop()
+                            
+                            try await model.saveKMLMultiTrack()
+                            multiTrack = try await model.loadKMLMultiTrack()
+                        case .reseting:
+                            model.reset()
+                            multiTrack = nil
+                            
+                            await mapViewProxy.setViewpointScale(model.locationDisplay.initialZoomScale)
+                            try await model.startNavigation()
+                        }
+                    } catch {
+                        self.error = error
+                    }
+                }
+                .task {
+                    // Starts the navigation when the sample opens.
+                    do {
+                        try await model.startNavigation()
+                    } catch {
+                        self.error = error
+                    }
+                    
+                    // Monitors the auto pan mode to determine if recenter button should be enabled.
+                    for await autoPanMode in model.locationDisplay.$autoPanMode {
+                        isRecenterEnabled = autoPanMode != .navigation
+                    }
+                }
+                .errorAlert(presentingError: $error)
+        }
+    }
+}
+
+private extension CreateKMLMultiTrackView {
+    /// A picker for selecting a track from a KML multi-track.
+    struct TrackPicker: View {
+        /// The KML tracks options shown in the picker.
+        let tracks: [KMLTrack]
+        
+        /// The closure to perform when the selected track has changed.
+        let onSelectionChanged: (Geometry) async -> Void
```

Comment:
This closure doesn't need to be `async`, as the "onValueChanged" semantics usually happen synchronously. Since this is an internal component, no need to change.